slapo/schedule.py

```diff
@@ -1366,14 +1362,24 @@ def _consolidate_and_broadcast(sch: Schedule):
                     axis = idx
                     is_found = True
             if is_found:
+                cnt_shard += 1
                 new_param = param.detach().split(sharded_size, dim=axis)[tp_rank]
                 sch.mod.register_parameter(param_name, nn.Parameter(new_param))
 
         for subsch in sch.child.values():
-            _consolidate_and_broadcast(subsch)
+            ret = _consolidate_and_broadcast(subsch)
+            num_params += ret[0]
+            cnt_shard += ret[1]
+
+        return num_params, cnt_shard
 
     if cnt_meta != 0 or cnt_materialized != 0:
-        _consolidate_and_broadcast(sch)
+        num_params, cnt_shard = _consolidate_and_broadcast(sch)
+
+    logger.info(
+        "Finished consolidating %d parameters with %d being sharded"
```

Comment:
"parameter tensors" is better to avoid confusion

Reply:
Updated.